R/add_difference_row.R

```diff
@@ -0,0 +1,435 @@
+#' Add difference rows
+#'
+#' - [`add_difference_row.tbl_summary()`]
+#'
+#' @param x (`gtsummary`)\cr
+#'   Object with class 'gtsummary'
+#' @param ... Passed to other methods.
+#' @keywords internal
+#'
+#' @author Daniel D. Sjoberg
+#' @export
+add_difference_row <- function(x, ...) {
+  check_not_missing(x)
+  check_class(x, "gtsummary")
+  UseMethod("add_difference_row")
+}
+
+#' Add differences rows between groups
+#'
+#' @description
+#' `r lifecycle::badge('experimental')`\cr
+#' Adds difference to tables created by [`tbl_summary()`] as additional rows.
+#' This function is often useful when there are more than two groups to compare.
+#'
+#' Pairwise differences are calculated relative to the specified
+#' `by` variable's specified reference level.
```

Comment:
```suggestion
#' Pairwise differences are calculated relative to the specified
#' `by` variable's reference level.
```

Reply:
I think it's ok to leave as is, since the reference level does need to be specified.